nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/AddEntryToMultiValuedPropertyTest.java

```diff
@@ -95,6 +98,52 @@ public void shouldAddEntryIntoStringList() throws Exception {
     }
 
     @Test
+    public void shouldAddDifferentEntryIntoList() throws Exception {
+        operation.xpath = "lists:double";
+        operation.value = 1.0;
+        operation.run(doc);
+        operation.value = 2.0;
+        operation.run(doc);
+        Double[] doubleValues = (Double[]) doc.getPropertyValue("lists:double");
+        assertEquals(2, doubleValues.length);
+        assertEquals(1.0, doubleValues[0]);
+        assertEquals(2.0, doubleValues[1]);
+
+        operation.xpath = "lists:boolean";
+        operation.value = true;
+        operation.run(doc);
+        operation.value = false;
+        operation.run(doc);
+        Boolean[] booleanValues = (Boolean[]) doc.getPropertyValue("lists:boolean");
+        assertEquals(2, booleanValues.length);
+        assertEquals(Boolean.TRUE, booleanValues[0]);
+        assertEquals(Boolean.FALSE, booleanValues[1]);
+
+        operation.xpath = "lists:integer";
+        operation.value = 1;
+        operation.run(doc);
+        operation.value = 2;
+        operation.run(doc);
+        Long[] longValues = (Long[]) doc.getPropertyValue("lists:integer");
+        assertEquals(2, longValues.length);
+        assertEquals(Long.valueOf(1), longValues[0]);
+        assertEquals(Long.valueOf(2), longValues[1]);
+
+        operation.xpath = "lists:date";
+        Calendar now1 = Calendar.getInstance();
+        sleep(1);
```

Comment:
You should sleep a tiny bit more, like 10ms, as some OS like Windows has a time granularity that's coarser than 1ms.